server/controllers/images.ts

```diff
@@ -0,0 +1,33 @@
+import express, { Request, Response, NextFunction } from "express";
+import { storage } from '../lib/firebase'
+import { ref, uploadBytes, getDownloadURL } from 'firebase/storage'
+import { HF_TOKEN } from "../utils/config";
+
+const imageRouter = express.Router();
+imageRouter.use(express.json());
+
+imageRouter.post("/", async (req: Request, res: Response) => {
+    try {
+        const response = await fetch(
```

Comment:
This error handling is almost there! We do still need to check for response.ok on this response, and throw an error if it's false. This confirms that the response was successful, because some unsuccessful requests do not automatically return an error (which can happen for errors 301, 400, 404, 403, 500, and more).

`if (!response.ok) {
    throw new Error("The response from the network was not ok");
}`

- https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful
- https://sangster.dev/posts/fetch-response-ok/